bin/install_swiftpm_dependencies

```diff
@@ -1,30 +1,91 @@
 #!/bin/bash -eu
 
-CUSTOM_PACKAGE_RESOLVED_PATH=${1-}
-BUILD_TYPE=${2-}
+print_usage() {
+  echo "Usage:"
+  echo "  Using the \`Package.resolved\` of an Xcode \`.xworkspace\`:"
+  echo "    $0 --workspace PATH --scheme NAME"
+  echo "  Using the \`Package.resolved\` of an Xcode \`.xcodeproj\`:"
+  echo "    $0 --project PATH"
+  echo "  Using the \`Package.resolved\` at the root, managed by \`swift package\` (and not Xcode)"
+  echo "    $0 --use-spm"
+}
+
+# Parse input parameters
+XCWORKSPACE_PATH=
+SCHEME_NAME=
+XCODEPROJ_PATH=
+USE_SPM=false
+while [[ "$#" -gt 0 ]]; do
+	case $1 in
+		--workspace)
+			XCWORKSPACE_PATH="$2"
+      shift
+			;;
+		--scheme)
+			SCHEME_NAME=$2
+      shift
+			;;
+    --project)
+      XCODEPROJ_PATH=$2
+      shift
+      ;;
+    --use-spm)
+      USE_SPM=true
+      ;;
+    -h|--help)
+      print_usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown option or unexpected argument: $1"
+      print_usage
+      exit 2
+      ;;
+	esac
+	shift
+done
+
 SPM_CACHE_LOCATION="$HOME/Library/Caches/org.swift.swiftpm"
 
-WORKSPACE_PACKAGE_RESOLVED_PATH=(*.xcworkspace/xcshareddata/swiftpm/Package.resolved)
-ROOT_PACKAGE_RESOLVED_PATH=Package.resolved
+# Try to guess what to do if no option provided explicitly
+if [[ -z "${XCWORKSPACE_PATH}" && -z "${XCODEPROJ_PATH}" && "${USE_SPM}" != "true" ]]; then
+  echo "No \`--workspace\`, \`--project\` or \`--use-spm\` flag provided. Trying to guess the correct one to use..."
+  FOUND_ROOT_WORKSPACE_PATHS=(*.xcworkspace)
+  if  [[ ${#FOUND_ROOT_WORKSPACE_PATHS[@]} -eq 1 && -d "${FOUND_ROOT_WORKSPACE_PATHS[0]}" && ! -f "Package.resolved" ]]; then
+    XCWORKSPACE_PATH="${FOUND_ROOT_WORKSPACE_PATHS[0]}"
+    echo " --> Found a single \`.xcworkspace\` file at the root of the repo (and no root \`Package.resolved\`)"
+    echo "     so we are assuming \`--workspace \"${XCWORKSPACE_PATH}\"\`."
```

Comment:
What do you think of:

```suggestion
    echo "--> Found a single \`.xcworkspace\` file and no \`Package.resolved\` in the root of the repo."
    echo "     Defaulting to \`--workspace \"${XCWORKSPACE_PATH}\"\`."
```

The pattern I'm suggesting here: splits the sentence in two, removing the "so", and move the "in the root of the repo" at the end of the sentence. _Just a nitpick_

Reply:
Fixed in e220c485bfa5fe22e861469399a4f1686ae5f371